core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2698,6 +2706,133 @@ public void testTableExistsFallbackToGETRequestWithLegacyServer() {
     verifyTableExistsFallbackToGETRequest(ConfigResponse.builder().build());
   }
 
+  @Test
+  public void testETagCreateTableAndLoadTable() {
+    Map<String, String> respHeaders = Maps.newConcurrentMap();
+
+    RESTCatalog catalog = setUpETagTest(respHeaders);
+
+    catalog.createTable(TABLE, SCHEMA);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    String eTag = respHeaders.get(HttpHeaders.ETAG);
+    respHeaders.clear();
+
+    catalog.loadTable(TABLE);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    assertThat(eTag).isEqualTo(respHeaders.get(HttpHeaders.ETAG));
+    respHeaders.clear();
+  }
+
+  @Test
+  public void testETagDifferentTables() {
+    Map<String, String> respHeaders = Maps.newConcurrentMap();
+
+    RESTCatalog catalog = setUpETagTest(respHeaders);
+
+    catalog.createTable(TABLE, SCHEMA);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    String eTagTbl1 = respHeaders.get(HttpHeaders.ETAG);
+    respHeaders.clear();
+
+    catalog.createTable(TableIdentifier.of(TABLE.namespace(), "table2"), SCHEMA);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    assertThat(eTagTbl1).isNotEqualTo(respHeaders.get(HttpHeaders.ETAG));
+  }
+
+  @Test
+  public void testETagAfterDataUpdate() {
+    Map<String, String> respHeaders = Maps.newConcurrentMap();
+
+    RESTCatalog catalog = setUpETagTest(respHeaders);
+
+    catalog.createTable(TABLE, SCHEMA);
+
+    respHeaders.clear();
+
+    Table tbl = catalog.loadTable(TABLE);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    String eTag = respHeaders.get(HttpHeaders.ETAG);
+    respHeaders.clear();
+
+    tbl.newAppend().appendFile(FILE_A).commit();
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    assertThat(eTag).isNotEqualTo(respHeaders.get(HttpHeaders.ETAG));
+  }
+
+  @Test
+  public void testETagAfterMetadataOnlyUpdate() {
+    Map<String, String> respHeaders = Maps.newConcurrentMap();
+
+    RESTCatalog catalog = setUpETagTest(respHeaders);
+
+    catalog.createTable(TABLE, SCHEMA);
+
+    respHeaders.clear();
+
+    Table tbl = catalog.loadTable(TABLE);
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    String eTag = respHeaders.get(HttpHeaders.ETAG);
+    respHeaders.clear();
+
+    tbl.updateSchema().addColumn("extra", Types.IntegerType.get()).commit();
+
+    assertThat(respHeaders).containsKey(HttpHeaders.ETAG);
+    assertThat(eTag).isNotEqualTo(respHeaders.get(HttpHeaders.ETAG));
+  }
+
+  @Test
+  public void testETagRegisterTable() {
+    Map<String, String> respHeaders = Maps.newConcurrentMap();
+
+    RESTCatalog catalog = setUpETagTest(respHeaders);
+
+    catalog.createTable(TABLE, SCHEMA);
```

Comment:
```suggestion
    Table tbl = catalog.createTable(TABLE, SCHEMA);
```

then you can remove the separate loading of the table further below